src/parsing/pretty.ml

```diff
@@ -245,6 +244,35 @@ let notation : Sign.notation pp = fun ppf n ->
   | Quant -> out ppf "quantifier"
   | _ -> ()
 
+(**[proof p ppf n] pretty prints the proof [p] using the formatter
+[ppf] at and identation level of [n] (initially 0)*)
```

Comment:
```ocaml
(** [proof ppf p] pretty prints proof [p] using formatter [ppf] *)
let rec proof : (p_proof * p_proof_end) pp =
  fun ppf (p, pe) ->
    out ppf "begin@[<hv2>@ %a@ @]%a"
      (fun ppf -> function
        (* No braces for a single toplevel block *)
      | [block] -> proof_steps ppf block
      | blocks -> proof_blocks ppf blocks) p
      proof_end pe
and proof_blocks : p_subproof list pp  =
  fun ppf blocks ->
    out ppf "@[<hv>%a@]"
      (pp_print_list ~pp_sep:pp_print_space proof_block) blocks
and proof_block : p_subproof pp =
  fun ppf block ->
    out ppf "{@[<hv2>@ %a@ @]}"
      proof_steps block
and proof_steps : p_proof_step list pp =
  fun ppf steps ->
    pp_print_list ~pp_sep:pp_print_space proof_step ppf steps
and proof_step : p_proof_step pp =
  fun ppf (Tactic (t, blocks)) ->
    out ppf "@[<hv2>%a@,%a;@];"
      tactic t
      proof_blocks blocks
```
Testing should be done to ensure I didn't do it wrong, but this should look decent.

All formatting should be redone so that no `\n` remain at all, they are fundamentally incompatible with the way `Forat` works